csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -618,14 +618,14 @@ public class OutgoingAsync : ProxyOutgoingAsyncBase
                              Ice.OutputStream? os = null, Ice.InputStream? iss = null) :
             base(prx, completionCallback, os, iss)
         {
-            Encoding = Protocol.GetCompatibleEncoding(Proxy.IceReference.GetEncoding());
```

Comment:
These methods provided an "auto-downgrade" support for minor versions. If you call a proxy with -p 1.1 and Ice understands only 1.0, Ice automatically calls it with protocol 1.0. For Ice 4.0, we're dropping minor protocol versions, so this becomes irrelevant.
There is the same feature for encoding: if you call a proxy with -e 1.2 and Ice understands only 1.1, Ice automatically calls it with encoding 1.1. For Ice 4.0, we're actually breaking this logic since we're dropping support for encoding 1.0 while still supporting 1.1. So going forward, we won't provide any auto-downgrade based on minor encoding versions.